### PR TITLE
Support CUDA 11.8

### PR DESCRIPTION
**Pull request by takagi**
#7080

### CODE REVIEW

File: cupy/_core/include/cupy/_cuda/README.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ https://docs.nvidia.com/cuda/archive/9.2/eula/#nvidia-cuda-toolkit-license-agree
 For CUDA 11.2+, we enable CUDA Enhanced Compatibility by hosting the fp16 headers from the latest
 CUDA Toolkit release and place them in the ``cuda-11`` folder.
 
-Currenly, the headers in ``cuda-11`` are from CTK 11.7.
+Currenly, the headers in ``cuda-11`` are from CTK 11.8.
```
Comment (kmaehashi) [inline cupy/_core/include/cupy/_cuda/README.md:8]:
```suggestion
Currenly, the headers in ``cuda-11`` are from CTK 11.8.0.
```

**Comment by kmaehashi** [comment]
Thanks @takagi!
* Regenerate needed.
* Could you create projects at https://ci.preferred.jp/ and run tests via the Web UI? Note that `/test` commands works with config on master branch for security reasons.

**Comment by kmaehashi** [comment]
Thanks! Now you can create a job for this PR:
https://ci.preferred.jp/cupy.linux.cuda118/
https://ci.preferred.jp/cupy.linux.cuda118.multi/

**Comment by takagi** [comment]
Will update NCCL in install_library.py in another PR as NCCL 2.15 is not found here:
https://developer.download.nvidia.com/compute/redist/nccl/v2.15/nccl_2.15.1-1+cuda11.8_x86_64.txz

**Comment by takagi** [comment]
Now the following tests passed!
https://ci.preferred.jp/cupy.linux.cuda118/
https://ci.preferred.jp/cupy.linux.cuda118.multi/

**Comment by kmaehashi** [comment]
LGTM!

**Pull request merged**